base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
And once you add `arr` to free category, you have two representations of identity morphisms. (Not necessarily a big deal, though, just pointing that out.) To avoid that, you could start with free `Compose`.